crates/sui-types/src/effects/effects_v2.rs

```diff
@@ -0,0 +1,503 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::object_change::{IDOperation, ObjectIn, ObjectOut};
+use super::EffectsObjectChange;
+use crate::base_types::{
+    EpochId, ObjectDigest, ObjectID, ObjectRef, SequenceNumber, SuiAddress, TransactionDigest,
+    VersionDigest,
+};
+use crate::digests::{Digest, TransactionEventsDigest};
+use crate::effects::{InputSharedObjectKind, TransactionEffectsAPI};
+use crate::execution_status::ExecutionStatus;
+use crate::gas::GasCostSummary;
+use crate::message_envelope::Message;
+use crate::object::{Owner, OBJECT_START_VERSION};
+use crate::transaction::SenderSignedData;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashSet};
+
+/// The response from processing a transaction or a certified transaction
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+pub struct TransactionEffectsV2 {
+    /// The status of the execution
+    status: ExecutionStatus,
+    /// The epoch when this transaction was executed.
+    executed_epoch: EpochId,
+    gas_used: GasCostSummary,
+    /// The transaction digest
+    transaction_digest: TransactionDigest,
+    /// The updated gas object reference, as an index into the `changed_objects` vector.
+    /// Having a dedicated field for convenient access.
+    /// System transaction that don't require gas will leave this as None.
+    gas_object_index: Option<u16>,
```

Comment:
I don't think its completely inconceivable we would want to exceed 2^16 outputs at some point (due to child objects), and given that this is a huge struct already (probably 200-300 bytes from eyeballing it), why not just make this a `u32` or even `usize`?